### PR TITLE
[SPARK-20704][SPARKR] change CRAN test to run single thread

### PR DESCRIPTION
**Pull request by felixcheung**
## What changes were proposed in this pull request?

- [x] need to test by running R CMD check --as-cran
- [x] sanity check vignettes

## How was this patch tested?

Jenkins

### CODE REVIEW

File: R/pkg/DESCRIPTION [hunk 1]
```diff
@@ -13,6 +13,7 @@ Authors@R: c(person("Shivaram", "Venkataraman", role = c("aut", "cre"),
 License: Apache License (== 2.0)
 URL: http://www.apache.org/ http://spark.apache.org/
 BugReports: http://spark.apache.org/contributing.html
+OS_Type: unix
```
Comment (shivaram) [inline R/pkg/DESCRIPTION]: Can we do this in a separate PR ? It'll be good to track this as a separate JIRA / PR just to lookup why we did this. Also I'll hopefully hear from the winbuilder maintainer if this is something we can avoid doing.

File: R/pkg/inst/tests/testthat/jarTest.R [hunk 2]
```diff
@@ -16,7 +16,7 @@
 #
 library(SparkR)
 
-sc <- sparkR.session()
+sc <- sparkR.session(master = "local[1]")
```
Comment (shivaram) [inline R/pkg/inst/tests/testthat/jarTest.R]: any specific reason these dont use `sparkRMaster` variable ?

File: R/pkg/inst/tests/testthat/test_Serde.R [hunk 3]
```diff
@@ -17,7 +17,7 @@
 
 context("SerDe functionality")
 
-sparkSession <- sparkR.session(enableHiveSupport = FALSE)
+sparkSession <- sparkR.session(master = sparkRMaster, enableHiveSupport = FALSE)
```
Comment (shivaram) [inline R/pkg/inst/tests/testthat/test_Serde.R]: Minor nit - Is `sparkRTestMaster` is a better name ?

**Review by shivaram — COMMENTED** [review]
Thanks @felixcheung -- I just had one major comment on pulling out the OS_Type into a separate PR

File: R/pkg/DESCRIPTION (hunk 1 above)
Comment (felixcheung) [inline R/pkg/DESCRIPTION]: right, I can revert this before merging. I want to have a branch to test devtools install

File: R/pkg/inst/tests/testthat/jarTest.R (hunk 2 above)
Comment (felixcheung) [inline R/pkg/inst/tests/testthat/jarTest.R]: it's not accessible here - this is a separate R app

Comment (shivaram) [inline R/pkg/inst/tests/testthat/jarTest.R]: Ah got it.

**Comment by shivaram** [comment]
Actually one more thing to do is to change the vignettes to use 1 core as well since they get rebuilt / checked during the CRAN check.

**Comment by felixcheung** [comment]
yes, it's here https://github.com/apache/spark/pull/17945/files#diff-56790892076daac0bc43a9b38c0b1470R51

**Comment by felixcheung** [comment]
tested devtools install on Windows (even with OS_type unix)

**Comment by felixcheung** [comment]
hmm, I'm getting a warning? from R CMD check
```
Unknown, possibly mis-spelled, fields in DESCRIPTION:
  ‘OS_Type’
```

**Comment by shivaram** [comment]
I think its a lowercase `t`

**Comment by shivaram** [comment]
Can you also retest with devtools after the fix ?

**Comment by felixcheung** [comment]
checked vignettes

**Comment by felixcheung** [comment]
bummer, no one can install on windows then
```
Downloading GitHub repo felixcheung/spark@rchangesforpackage
from URL https://api.github.com/repos/felixcheung/spark/zipball/rchangesforpackage
Installing SparkR

ERROR:  Unix-only package
Error: Command failed (1)

```

**Comment by shivaram** [comment]
Ah that is a bummer. Lets hold off on that and use a new JIRA to discuss it.  Rest of the changes LGTM pending jenkins, appveyor

**Comment by felixcheung** [comment]
AppVeyor was failing because of the `SparkListenerBus` issue.

**Comment by felixcheung** [comment]
merged to master/2.2

**Pull request closed without merging**